ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -395,18 +424,42 @@ inline MatmulProgramConfig create_simple_matmul_program_config(
                 (use_mcast_config and mem_config.memory_layout == TensorMemoryLayout::BLOCK_SHARDED)) {
             bool transpose_mcast = input_tensor_a.memory_config().memory_layout == TensorMemoryLayout::BLOCK_SHARDED &&
                                    input_tensor_a.shard_spec().value().orientation == ShardOrientation::COL_MAJOR;
+            uint32_t out_block_h = per_core_M;
+            uint32_t out_block_w = per_core_N;
             out_subblock_h = 4;
             out_subblock_w = 2;
             if (out_subblock_w != per_core_N) {
                 out_subblock_h = 1;
             }
+            if (input_tensor_a.memory_config().memory_layout == TensorMemoryLayout::INTERLEAVED
+                && mem_config.memory_layout == TensorMemoryLayout::INTERLEAVED
+                && input_tensor_b.memory_config().memory_layout == TensorMemoryLayout::INTERLEAVED
+                && input_tensor_a.memory_config().buffer_type == BufferType::DRAM
+                && input_tensor_b.memory_config().buffer_type == BufferType::DRAM
+                && mem_config.buffer_type == BufferType::DRAM
+                && num_cores_x == 8 && num_cores_y == 8) {
+
+                in0_block_w = !transpose_mcast ? Kt / num_cores_x : Kt / num_cores_y;
+                per_core_M = !transpose_mcast ? Mt / num_cores_y : Mt / num_cores_x;
+                per_core_N = !transpose_mcast ? Nt / num_cores_x : Nt / num_cores_y;
+
+                auto mutlti_dim_per_core_factor = get_mutlti_dim_per_core_factor(input_tensor_a, input_tensor_b, per_core_M, per_core_N, in0_block_w, tt_metal::detail::TileSize(tt::DataFormat::Float16_b));
```

Comment:
what does `mutlti` mean? initially i thought that it's a typo, but probably it does mean something I don't know :)

Reply:
typo, thanks for catching this